### ISSUE TITLE
`nix search` never finds any packages

### ISSUE THREAD

**Issue by lilyball** [NixOS/nix#1892]
The brand new `nix search` command in Nix 2.0 doesn't seem to work. Nothing I give it will result in actually finding any packages. It doesn't matter if I use the `--no-cache` flag or the `--update-cache` flag or no flag at all. In fact, it appears that trying to update the cache also wrote an empty cache file, because after using that flag, subsequent attempts at searching give me an error "expected JSON value".

 - system: `"x86_64-darwin"`
 - host os: `Darwin 17.4.0, macOS 10.13.3`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.0`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

**Comment by dtzWill** [comment]
Is your NIX_PATH empty, by chance? Not sure what'd cause this otherwise :(.

And to be clear, does "nix search hello" turn up any results?

Also just submitted PR to fix the error (writing empty/invalid cache with `--no-cache`) as well as the empty search string problem reported in #1893.

**Comment by kballard** [comment]
My `NIX_PATH` is `/nix/var/nix/profiles/per-user/root/channels`, and `nix search --no-cache hello` immediately returns with no results. The empty return is fast enough I have to assume it's not actually searching anything.

```
> time nix search --no-cache hello
        0.01 real         0.00 user         0.00 sys
> 
```

**Comment by clhodapp** [comment]
My own finding (on another mac) was that `nix search` didn't work while I had `nixpgs=https://github.com/NixOS/nixpkgs-channels/archive/nixpkgs-unstable.tar.gz` on my `NIX_PATH` but DOES work when I point `nixpkgs` at a local path.

**Comment by kballard** [comment]
When you say "at a local path", you mean when you clone the `nixpkgs` repo yourself? I haven't done that, I'm still using the default.

In any case, I really want fast searching, so if anyone needs me to do some debugging, just let me know what you want and I'll do it. Just remember I have very little familiarity with how Nix works so e.g. I don't even know how to point `nixpkgs` at a local path.

**Comment by kballard** [comment]
Looks like setting `NIX_PATH` to a dummy value like `NIX_PATH='nixpkgs=asdf:/nix/var/nix/profiles/per-user/root/channels'` makes `nix search` work. So presumably same root cause as #1865.

**Comment by jcrben** [comment]
Noticed this case with postgrest just now - wonder if it might have to do with attribute name not being searched tho?

![image](https://user-images.githubusercontent.com/5614134/39670374-19f995e6-50b8-11e8-9af6-a90a467c5032.png)

I also couldn't pull it up with nix-env:
![image](https://user-images.githubusercontent.com/5614134/39670383-4c23e51c-50b8-11e8-8465-c16e879cde68.png)

**Comment by ElvishJerricco** [comment]
Should this be closed as a duplicate of #1865, since they're same root cause, as @kballard mentioned?

**Comment by kballard** [comment]
I think it should.
